plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationProviderImpl.kt

```diff
@@ -152,14 +137,36 @@ internal class LocationProviderImpl(context: Context, var settings: LocationComp
   override fun unRegisterLocationConsumer(locationConsumer: LocationConsumer) {
     locationConsumers.remove(locationConsumer)
     if (locationConsumers.isEmpty()) {
-      locationEngine.removeLocationUpdates(this)
+      locationEngine.removeLocationUpdates(locationEngineCallback)
       handler?.removeCallbacks(runnable)
+      locationCompassEngine.removeCompassListener(this)
     }
   }
 
   private companion object {
     private const val TAG = "MapboxLocationProvider"
     private const val INIT_UPDATE_DELAY = 100L
     private const val MAX_UPDATE_DELAY = 5000L
+
+    private class CurrentLocationEngineCallback(locationProviderImpl: LocationProviderImpl) :
```

Comment:
class declared inside companion object? is it intended?

Reply:
Yes. Do you have any concerns about this？

Reply:
what's the benefit of it? comparing to inner class or similar

Reply:
should we test this class? 
and maybe make it as internal class in this file, not in `private companion`?

Reply:
Moved to an internal class. The current test cases could cover this class as it's in the route when a new location is updated.